packages/commerce-generic/src/services/cart/discount.ts

```diff
@@ -0,0 +1,87 @@
+import { Cart, Voucher, Promotion } from '@composable/types'
+import { centToString, toCent } from './to-cent'
+
+const examplePromotion: Promotion = {
+  id: 'prom_1',
+  label: 'Black Friday 2024 - 10$',
+  discountAmount: '10',
+}
+
+const vouchersAvailable: Voucher[] = [
+  { code: '5$OFF', label: '5 bucks off in winter', discountAmount: '5' },
+  { code: '15$OFF', label: '15 bucks off in winter', discountAmount: '15' },
+]
+
+export const deleteVoucherFromCart = async (
+  cart: Cart,
+  code: string
+): Promise<{ cart: Cart; success: boolean; errorMessage?: string }> => {
+  const success = true
+  const errorMessage = undefined
+  const updatedCart = await updateCartDiscount({
+    ...cart,
+    vouchersApplied: [
+      ...(cart.vouchersApplied?.filter((voucher) => voucher.code !== code) ||
+        []),
+    ],
+  })
+  return {
+    cart: updatedCart,
+    success,
+    errorMessage,
+  }
+}
+
+export const addVoucherToCart = async (
+  cart: Cart,
+  code: string
+): Promise<{ cart: Cart; success: boolean; errorMessage?: string }> => {
+  if (cart.vouchersApplied?.some((voucher) => voucher.code === code)) {
+    return {
+      cart,
+      success: false,
+      errorMessage: 'Voucher is already applied',
+    }
+  }
+  const voucher = vouchersAvailable.find((voucher) => voucher.code === code)
+  if (!voucher) {
+    return {
+      cart,
+      success: false,
+      errorMessage: 'Voucher not found',
+    }
+  }
+  const success = true
+  const errorMessage = undefined
+  const updatedCart = await updateCartDiscount({
+    ...cart,
+    vouchersApplied: [...(cart.vouchersApplied || []), voucher],
+  })
+  return {
+    cart: updatedCart,
+    success,
+    errorMessage,
+  }
+}
+
+export const updateCartDiscount = async (cart: Cart): Promise<Cart> => {
```

Comment:
Thank you for this suggestion. I also changed it from the `examplePromotion` (object) to `examplePromotions` (array of objects). I think it would be better to be able to add more than one promotion. But, of course, you can choose not to use them at all.